### PR TITLE
feat(role): Add retirement flag for org roles

### PR DESCRIPTION
**Pull request by RyanSkonnord**
Sub-feature to https://github.com/getsentry/sentry/pull/32498.

This allows an organization role to be removed from the UI without migrating it out of the database. All members with the retired role will keep it and all associated permissions, but the role can't be assigned.

Role retirement is behind the "team-roles" feature flag. The retired role remains visible for organizations without the flag.

Retire the org "Admin" role, expecting "Team Admin" to replace it.

### CODE REVIEW

**Review by wedamija — COMMENTED** [review]
Will `team-admin` have different permissions to `admin`? If not, is it better to just rename in the db?

**Comment by wedamija** [comment]
> Will `team-admin` have different permissions to `admin`? If not, is it better to just rename in the db?

Ok, just read the linked pr a bit, so we're going to allow roles per team. Since `admin` is already scoped to teams you're a member of, any reason we couldn't use the same role there?

File: src/sentry/api/serializers/models/role.py [hunk 1]
```diff
@@ -11,5 +11,6 @@ def serialize(self, obj, attrs, user, **kwargs):
             "desc": obj.desc,
             "scopes": obj.scopes,
             "is_global": obj.is_global,
+            "is_retired": obj.is_retired,
```
Comment (markstory) [inline src/sentry/api/serializers/models/role.py:14]: Should we use camelBacked parameters like the bulk of the API does? I recognize that we will be adding a small inconsistency with `is_global` but we could also add `isGlobal` to rectify that.

**Review by nhsiehgit — APPROVED** [review]

File: src/sentry/api/serializers/models/role.py (hunk 1 above)
Comment (RyanSkonnord) [reply src/sentry/api/serializers/models/role.py]: Done.

**Comment by RyanSkonnord** [comment]
> Since `admin` is already scoped to teams you're a member of, any reason we couldn't use the same role there?

The idea is that we eventually want to get rid of "admin" as an org-wide thing, and present it only as an option as a per-team basis. Flipping this switch would remove the option for new members as the first step. Later steps would entail changing the set of scopes given by Org Admin versus Team Admin, and possibly a data migration to remove the role from existing members. (Details in ER-981)

**Comment by wedamija** [comment]
> > Since `admin` is already scoped to teams you're a member of, any reason we couldn't use the same role there?
> 
> The idea is that we eventually want to get rid of "admin" as an org-wide thing, and present it only as an option as a per-team basis. Flipping this switch would remove the option for new members as the first step. Later steps would entail changing the set of scopes given by Org Admin versus Team Admin, and possibly a data migration to remove the role from existing members. (Details in ER-981)

Ok, I think that this should probably be discussed at a wider level than just your team. This results in a fairly large change to how roles are set up today - so there's no way to say "this user is an admin on all teams they belong to" anymore? I think this results in potentially a lot more work. Does it make sense to just make this an overridable thing? So you still have your org level role (member, admin, etc), and you can override it on a specific team.

**Comment by RyanSkonnord** [comment]
@wedamija Thanks for the input. I'll follow up over Slack.

**Comment by RyanSkonnord** [comment]
Closing along with #32498; will reopen after more prerequisite steps are out of the way.

**Pull request closed without merging**